src/streamlit_extras/__init__.py

```diff
@@ -0,0 +1,45 @@
+import inspect
+from importlib import import_module
+from typing import Any, Callable, Optional, TypeVar, Union, overload
+
+from streamlit import _gather_metrics
+
+F = TypeVar("F", bound=Callable[..., Any])
+
+
+@overload
+def extra(
+    func: F,
+) -> F:
+    ...
+
+
+@overload
+def extra(
+    func: None = None,
+) -> Callable[[F], F]:
+    ...
+
+
+def extra(
+    func: Optional[F] = None,
+) -> Union[Callable[[F], F], F]:
+
+    if func:
+
+        filename = inspect.stack()[1].filename
+        extra_name = "streamlit_extras." + filename.split("/")[-2]
+        module = import_module(extra_name)
+
+        if hasattr(module, "__funcs__"):
+            module.__funcs__ += [func]  # type: ignore
+        else:
+            module.__funcs__ = [func]  # type: ignore
+
+        name = f"{module}.{func.__name__}"
+        return _gather_metrics(name=name, func=func)
```

Comment:
🎉

Reply:
@arnaudmiribel Since there isn't a try catch fallback here, you probably need to pin Streamlit to version `>=1.14`. Streamlit versions below `1.14` doe not have `_gather_metrics` with `name` parameter.

Reply:
That's a good point. I'm fine with requiring latest streamlit - wdyt @blackary? - if not then we should try/except and just return the actual func.

Reply:
@arnaudmiribel I'd prefer slightly more loose restrictions, and just failing to gather metrics if it's too old

Reply:
Added a try/except 8392df5